src/IROperator.h

```diff
@@ -1470,8 +1470,12 @@ namespace Internal {
  * context-dependent, because 'value' might be statically bounded at
  * some point in the IR (e.g. due to a containing if statement), but
  * not elsewhere.
+ *
+ * If pure = true, the call is marked as a PureIntrinsic, allowing hoisting
+ * of the value; the caller must be (even more) vigilant in this case to
+ * ensure that the promise will remain valid in such conditions.
  **/
-Expr promise_clamped(const Expr &value, const Expr &min, const Expr &max);
```

Comment:
I think promise_clamped is always pure if you use the definition I gave offline, and you don't need the pure flag:

"This intrinsic evaluates to its first argument. If this value is used by a side-effecting operation and it is outside the range specified by its second and third arguments, behavior is undefined. The compiler can therefore assume that the value is within the range given and optimize accordingly. Note that this permits promise_clamped to evaluate to something outside of the range, provided that this value is not used."

Reply:
That makes it legal to hoist.

Reply:
So are you suggesting that we should change the body of `promise_clamped()` to always use PureIntrinsic?

Reply:
Yes

Reply:
PTAL